app/javascript/components/time_table/project_reports/edit_report.js

```diff
@@ -189,11 +189,28 @@ export default class EditReport extends React.Component {
     });
   }
 
-  renderMergeButton(category, edit) {
+  onShowEdit(event, category) {
+    this.setState(({ currentBody }) => {
+      currentBody[category] = currentBody[category].map((wt) => {
```

Comment:
You are mutating state which is not allowed, antipattern . Spread it to create shallow copy of state and then return modified one.